src/app/zap-templates/templates/app/helper.js

```diff
@@ -423,6 +389,68 @@ function asTypedLiteral(value, type)
   return templateUtil.templatePromise(this.global, promise)
 }
 
+async function asTypedExpression(value, type)
+{
+  const valueIsANumber = !isNaN(value);
+  if (!value || valueIsANumber) {
+    return asTypedLiteral.call(this, value, type);
+  }
+
+  const tokens = value.split(' ');
+  if (tokens.length < 2) {
+    return asTypedLiteral.call(this, value, type);
+  }
+
+  const resultType = await asNativeType.call(this, type);
+  if (!TestHelper.chip_tests_variables_has.call(this, tokens[0])) {
+    return `static_cast<${resultType}>(${value})`;
+  }
+
+  const isNullable = TestHelper.chip_tests_variables_is_nullable.call(this, tokens[0]);
+  if (isNullable) {
+    value = `(${value}).Value()`;
```

Comment:
I don't understand this.  Shouldn't we be doing this even if there is only one token, if that token happens to be a variable?  And shouldn't we be appending `.Value()` to the nullable-variable token, as opposed to the whole value string?

Reply:
The reason is because I would like to avoid failing on reading the value of an empty nullable. My understanding is that `chip::Nullable` has adopted the same behavior than `chip::Optional` and throw if there if no value has been defined.

I would prefer to handle the error in the test suite.
The missing bits for making this part of the code clearer are probably https://github.com/vivien-apple/connectedhomeip-1/commit/8dae3e676cd55a9fc3616d14406f50ba452ddee7

Reply:
No, my point is that of `value` is something like `FooVariable + 1` where `FooVariable` is a nullable variable, then this will produce `(FooVariable + 1).Value()` which seems wrong.

And if it's just `FooVariable` then we will have returned earlier due to only having a single token, so won't be doing this `Value()` thing.

Reply:
I have removed the bits related to `isNullable` as discussed offline.